pkg/runtime/conversion_generator.go

```diff
@@ -302,11 +302,15 @@ func (g *conversionGenerator) generateConversionsForSlice(inType, outType reflec
 }
 
 func (g *conversionGenerator) generateConversionsForStruct(inType, outType reflect.Type) error {
+	errs := []string{}
```

Comment:
OK - sorry. I was still thinking about something different.

My understanding of what you are trying to achieve is that you would like to create a conversion function for the "outer" type, which converts only the fields that exists in both. This is what you will not achieve.

Now I understand that it's not what you wanted to achieve.

Sorry - I thought you were trying to achieve something different.

That LGTM